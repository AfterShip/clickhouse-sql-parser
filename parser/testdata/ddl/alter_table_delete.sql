ALTER TABLE test.events DELETE WHERE created_at < '2023-01-01';
