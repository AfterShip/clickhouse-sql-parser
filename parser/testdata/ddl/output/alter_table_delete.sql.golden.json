[
  {
    "AlterPos": 0,
    "StatementEnd": 61,
    "TableIdentifier": {
      "Database": {
        "Name": "test",
        "QuoteType": 1,
        "NamePos": 12,
        "NameEnd": 16
      },
      "Table": {
        "Name": "events",
        "QuoteType": 1,
        "NamePos": 17,
        "NameEnd": 23
      }
    },
    "OnCluster": null,
    "AlterExprs": [
      {
        "DeletePos": 24,
        "StatementEnd": 61,
        "WhereClause": {
          "LeftExpr": {
            "Name": "created_at",
            "QuoteType": 1,
            "NamePos": 37,
            "NameEnd": 47
          },
          "Operation": "\u003c",
          "RightExpr": {
            "LiteralPos": 51,
            "LiteralEnd": 61,
            "Literal": "2023-01-01"
          },
          "HasGlobal": false,
          "HasNot": false
        }
      }
    ]
  }
]